ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/text/TextComponentViewVariablesTests.kt

```diff
@@ -0,0 +1,843 @@
+            title = "Annual (App name)",
+            price = Price(
+                amountMicros = 200_000_000,
+                currencyCode = "MXN",
+                formatted = "$ 200",
+            ),
+            description = "Annual",
+            period = Period(value = 1, unit = Period.Unit.YEAR, iso8601 = "P1Y"),
+            freeTrialPeriod = Period(value = 1, unit = Period.Unit.MONTH, iso8601 = "P1M"),
+        )
+        private val productWithUpperCaseName = productYearlyUsdTwoOffers.copy(name = "ANNUAL")
+        private val productWithLowerCaseName = productYearlyUsdTwoOffers.copy(name = "annual")
+
+        @Suppress("DEPRECATION")
+        private val packageYearlyUsdTwoOffers = Package(
+            packageType = PackageType.ANNUAL,
+            identifier = "package_yearly",
+            offering = OFFERING_ID,
+            product = productYearlyUsdTwoOffers,
+        )
+
+        @Suppress("DEPRECATION")
+        private val packageMonthlyUsdOneOffer = Package(
+            packageType = PackageType.ANNUAL,
+            identifier = "package_monthly",
+            offering = OFFERING_ID,
+            product = productMonthlyUsdOneOffer,
+        )
+
+        @Suppress("DEPRECATION")
+        private val packageYearlyMxnOneOffer = Package(
+            packageType = PackageType.ANNUAL,
+            identifier = "package_yearly",
+            offering = OFFERING_ID,
+            product = productYearlyMxnOneOffer,
+        )
+
+        private val packageWithUpperCaseName = packageYearlyUsdTwoOffers.copy(product = productWithUpperCaseName)
+        private val packageWithLowerCaseName = packageYearlyUsdTwoOffers.copy(product = productWithLowerCaseName)
+
+        @Suppress("LongMethod", "Unused", "CyclomaticComplexMethod")
+        @JvmStatic
+        @ParameterizedRobolectricTestRunner.Parameters(name = "{0} = {2}")
+        fun parameters(): Collection<*> = Variable.values().map { variableName ->
+            when (variableName) {
+                Variable.PRODUCT_CURRENCY_CODE -> arrayOf(
+                    "{{ ${variableName.identifier} }}",
+                    Args(
+                        packages = listOf(packageYearlyUsdTwoOffers),
+                        locale = "en_US",
+                        storefrontCountryCode = "US",
+                        variableLocalizations = variableLocalizationKeysForEnUs(),
+                    ),
+                    "USD"
+                )
+
+                Variable.PRODUCT_CURRENCY_SYMBOL -> arrayOf(
+                    "{{ ${variableName.identifier} }}",
+                    Args(
+                        packages = listOf(packageYearlyUsdTwoOffers),
+                        locale = "en_US",
+                        storefrontCountryCode = "US",
+                        variableLocalizations = variableLocalizationKeysForEnUs(),
+                    ),
+                    "\$"
+                )
+
+                Variable.PRODUCT_PERIODLY -> arrayOf(
```

Comment:
Should we add tests for lifetime/custom packages with some of these variables? I believe many of them just return null... But might be better to make sure with tests.